horace_core/sqw/@Experiment/set_instrument.m

```diff
@@ -0,0 +1,86 @@
+function   obj = set_instrument(obj,instr_or_fun,varargin)
+% add or reset instrument, related to the given experiment class
+%
+if isa(instr_or_fun,'IX_inst')
+    if numel(instr_or_fun) == 1
+        obj.instruments.unique_objects = instr_or_fun;
+        if obj.instruments.n_runs ~= obj.n_runs
+            inst = obj.instruments;
+            inst = inst.expand_runs(obj.n_runs); % will work only for single unique instrument
+            obj.instruments = inst;
+        end
+    else
+        inst = obj.instruments;
+        if numel(instr_or_fun) ~= inst.n_runs
+            error('HORACE:Experiment:invalid_argument',...
+                'Multiple instruments (N=%d) provided as input for set_instrument function but the number is not 1 and not equal to the number of runs defined in container (%d)', ...
+                numel(instr_or_fun),inst.n_runs)
+        end
+        n_runs = inst.n_runs;
+        for i=1:n_runs
+            inst(i) = instr_or_fun(i);
+        end
+        obj.instruments = inst;
```

Comment:
What is the purpose of this?

Reply:
for compressing duplicated instruments into unique_objects_container. Actually, with my modifications to container, it would work through add.